hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/preloader/NearCachePreloader.java

```diff
@@ -157,16 +153,11 @@ private boolean checkHeader(BufferingInputStream bis) throws IOException {
     }
 
     /**
-     * Stores the keys from the supplied {@link NearCacheRecordMap} instances.
-     *
-     * We need to support multiple records maps here, since Hazelcast Enterprise has a segmented
-     * {@link com.hazelcast.internal.nearcache.NearCacheRecordStore} which contains multiple records maps.
+     * Stores the Near Cache keys from the supplied iterator.
      *
-     * @param records the {@link NearCacheRecordMap} instances to retrieve the keys from
-     * @param <R>     the {@link NearCacheRecord} type
-     * @param <NCRM>  the {@link NearCacheRecordMap} type
+     * @param keySetIterator the key set to store
      */
-    public <R extends NearCacheRecord, NCRM extends NearCacheRecordMap<KS, R>> void storeKeys(NCRM... records) {
+    public void storeKeys(Iterator keySetIterator) {
```

Comment:
Because of overly complex generics in the abstractions of the `NearCacheRecordStore` implementations: `AbstractNearCacheRecordStore<K, V, KS, R extends NearCacheRecord, NCRM extends NearCacheRecordMap<KS, R>>`. I eventually gave up to provide the correct type here in the scope of this PR.